src/resource/solidDataset.ts

```diff
@@ -827,6 +827,22 @@ export async function deleteContainer(
   }
 }
 
+function isChildResource(a: string, b: string): boolean {
+  const parent = new URL(b);
+  const child = new URL(a);
+  // Explicitly test on the whole URL to enforce similar origins.
+  const isAncestor = child.href.startsWith(parent.href);
+  const urlDiff = child.pathname.substring(
+    parent.pathname.length,
+    child.pathname.length
+  );
+  // The child path component that isn't present in the parent should only
+  // potentially include slashes at the end (if it is a container).
+  const isDirectChild =
+    urlDiff !== "" && urlDiff.includes(urlDiff.replace(/\//g, " ").trim());
```

Comment:
I still find this less intuitive and more buggy than https://github.com/inrupt/solid-client-js/pull/2059#discussion_r1248579674.

In particular; it looks like `isDirectChild` will incorrectly return true when `urlDiff` is the following values `/`, `//`, `///` etc; `/ /a//`. You could even do

```ts
if (urlDiff.startsWith('/')) urlDiff = urlDiff.slice(1)
if (urlDiff.endsWith('/')) urlDiff = urlDiff.slice(0, -1)
isDirectChild = urlDiff !== "" && !urlDiff.includes("/")
```

Secondly should we check to make sure that `search` attribute of both `URL`s is the empty string.

Reply:
You're right, it is cleaner, implemented in 3b5aa3c26786c23715c614c509c303fb08ad98f7.

To your second point, I don't know if the specification says anything about search params. There probably won't be search params in most cases, but if the spec doesn't explicitly prevent it let's not make assumptions.